### PR TITLE
Reset search state when extra params change

### PR DESCRIPTION
**Pull request by pmareke**
EX-4554

### CODE REVIEW

**Comment by pmareke** [comment]
```
 <!-- Extra params components -->
    <RenderlessExtraParam #default="{ value, updateValue }" name="warehouse" defaultValue="1234">
      <BaseDropdown @change="updateValue" :value="value" :items="items">
        <template #item="{ item, isHighlighted, isSelected }">
          <CheckIcon v-if="isSelected" />
          <ChevronLeftIcon v-if="isHighlighted" />
          Warehouse: {{ item }}
        </template>
      </BaseDropdown>
    </RenderlessExtraParam>
    <RenderlessExtraParam #default="{ value, updateValue }" name="catalog" defaultValue="en">
      <BaseDropdown @change="updateValue" :value="value" :items="['en', 'es']">
        <template #item="{ item, isHighlighted, isSelected }">
          <CheckIcon v-if="isSelected" />
          <ChevronLeftIcon v-if="isHighlighted" />
          Catalog: {{ item }}
        </template>
      </BaseDropdown>
    </RenderlessExtraParam>
```

File: packages/x-components/src/x-modules/search/wiring.ts [hunk 1]
```diff
@@ -162,5 +176,11 @@ export const searchWiring = createWiring({
   },
   UserReachedResultsListEnd: {
     increasePageAppendingResults
+  },
+  ExtraParamsChanged: {
```
Comment (davidmfempathy) [inline packages/x-components/src/x-modules/search/wiring.ts:180]: Do we have to reset the rest of entities within the module? results, partialResults, relatedTags, banners, promoteds, selectedFilters, spellcheckedQuery ?

**Review by davidmfempathy — CHANGES_REQUESTED** [review]

**Review by davidmfempathy — DISMISSED** [review]

File: packages/x-components/src/x-modules/search/wiring.ts (hunk 1 above)
Comment (pmareke) [reply packages/x-components/src/x-modules/search/wiring.ts:180]: I think as the request is send again this entities will be overridden right?

File: packages/x-components/src/x-modules/search/wiring.ts [hunk 2]
```diff
@@ -84,6 +84,13 @@ export const setSort = wireCommit('setSort');
  */
 export const setPage = wireCommit('setPage');
 
+/**
+ * Sets the search state `page`.
```
Comment (CachedaCodes) [inline packages/x-components/src/x-modules/search/wiring.ts:88]: Correct this doc.

```suggestion
 * Sets the search state `params`.
```

- [ ] Correct doc

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/search/wiring.ts (hunk 1 above)
Comment (davidmfempathy) [reply packages/x-components/src/x-modules/search/wiring.ts:180]: As we've talked through Slack, you're right.

**Review by guillei10 — DISMISSED** [review]

**Review by davidmfempathy — APPROVED** [review]

**Review by guillei10 — APPROVED** [review]

**Pull request merged**